src/encoded/static/components/dataset.js

```diff
@@ -1270,7 +1356,20 @@ const basicTableColumns = {
     },
 };
 
-const geneSilencingSeriesTableColumns = {
+
+/**
+ * Collect all donor objects from all the given biosamples. Any duplicate donors get removed.
+ * @param {array} biosamples List of biosamples from which to extract donors.
+ * @returns {array} List of all distinct donors.
+ */
+const collectDonorsFromBiosamples = (biosamples) => (
+    _(biosamples.reduce((donors, biosample) => (
+        biosample.donor ? donors.concat(biosample.donor) : donors
+    ), [])).uniq((donor) => donor.accession)
```

Comment:
Minor detail. Would it be a good idea to use Set or _.uniq and not both interchangably?

Reply:
I don’t know of a way to use sets with objects, only strings.